<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Smoothing</class>
 <widget class="QDockWidget" name="Smoothing">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>467</width>
    <height>315</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Smoothing</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="Mesh_groupBox">
      <property name="title">
       <string>Mesh</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QPushButton" name="angle_button">
         <property name="text">
          <string>Angle Smoothing</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>#Iterations: </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="angles_iter_spinBox"/>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="area_button">
         <property name="text">
          <string>Area Smoothing</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>#Iterations: </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="areas_iter_spinBox"/>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <widget class="QCheckBox" name="projection_checkBox">
         <property name="text">
          <string> Re-project Vertices</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Precision:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="precision_spinBox">
           <property name="decimals">
            <number>7</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="Shape_groupBox">
      <property name="title">
       <string>Shape</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QPushButton" name="curvature_flow_button">
         <property name="text">
          <string>Smooth Along Curvature Flow</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>#Iterations: </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="curvature_iter_spinBox"/>
         </item>
        </layout>
       </item>
       <item row="0" column="2">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="time_step_label">
           <property name="text">
            <string>Time Step:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="time_step_spinBox">
           <property name="decimals">
            <number>6</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <widget class="QCheckBox" name="explicit_checkBox">
         <property name="text">
          <string>Use Normalized Scheme</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QCheckBox" name="border_button">
         <property name="text">
          <string>Constrain Border Vertices</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
